src/main/webapp/WEB-INF/jsp/user-page.jsp

```diff
@@ -57,7 +57,10 @@
             <label for="username-input" class="col-sm-2 col-form-label">Username</label>
             <div class="col-sm-10">
               <input type="text" class="form-control" name="username" id="username-input" placeholder="Enter a username"
-                required>
+                required pattern="^[a-zA-Z0-9_@\.]*$" maxlength="16">
```

Comment:
are you going to make another PR for the backend side validation?

Reply:
ok will add to this PR